### DISCUSSION TITLE
disabled tls occurring at startup regardless of settings

### DISCUSSION THREAD

**Discussion by arbourp** [photoprism/photoprism#4325] [General]
I have set the configuration for PhotoPrism to use TLS, it has no problems creating the self signed certificate, however whenever I start the application the log indicate that TLS is disabled and there is no way to force it to not respond on non-TLS, and requests for the page using https ends up with a failure 

Here is the logs:

photoprism-1  | time="2024-06-09T16:47:55Z" level=debug msg="migrate: 20240112-000001 skipped"
photoprism-1  | time="2024-06-09T16:47:55Z" level=debug msg="migrate: completed in 38.627047ms"
photoprism-1  | time="2024-06-09T16:47:55Z" level=info msg="server: enabled gzip compression"
photoprism-1  | time="2024-06-09T16:47:55Z" level=info msg="webdav: shared /originals/"
photoprism-1  | time="2024-06-09T16:47:55Z" level=info msg="webdav: shared /import/"
**photoprism-1  | time="2024-06-09T16:47:55Z" level=info msg="server: disabled tls"**
photoprism-1  | time="2024-06-09T16:47:55Z" level=info msg="server: listening on 0.0.0.0:2342 [2.830575ms]"

**Comment by arbourp** [comment]
PHOTOPRISM_DISABLE_TLS: "false"                # disables HTTPS/TLS even if the site URL starts with https:// and a certificate is available
      PHOTOPRISM_DEFAULT_TLS: "true"                 # defaults to a self-signed HTTPS/TLS certificate if no other certificate is available

**Reply by lastzero** [reply]
https://docs.photoprism.app/getting-started/using-https/#2-self-signed-certificate

**Reply by arbourp** [reply]
I've got all the settings correct, or at least I think so.
PHOTOPRISM_INIT: "https"
PHOTOPRISM_DISABLE_TLS: "false"
PHOTOPRISM_DEFAULT_TLS: "true" 

No matter what it doesn't enable TLS.
I'm going to scan for other typos.

**Reply by lastzero** [reply]
What about the URL?

**Reply by arbourp** [reply]
I found I had mistyped the URL value, and forgot to include the 's' in https.
It is now providing a different error:
photoprism-1  | time="2024-06-09T17:19:24Z" level=info msg="server: disabled auto tls"
and now it seems to favour HTTPS but is not providing a certificate in response to the requests.

**Reply by arbourp** [reply]
I am forwarding the port through my router if that is relevant at all, so the query at the firewall is on a different port.
I'm going to align them and see what happens.

**Reply by lastzero** [reply]
The protocol, not the port.

**Reply by arbourp** [reply]
Here's what I've done so far, with no love on TLS.
It is no longer responding with HTTP, which is positive.  I am using port 9002 because my firewall may or may not allow the correct port that i want through.(It is a residential gateway with 'security'

Here's what I just set
ports: 
  - "9002:9002" 
  PHOTOPRISM_HTTP_PORT: "9002"
   PHOTOPRISM_SITE_URL: "https://myhost.dedyn.io:9002/"  
   PHOTOPRISM_DISABLE_TLS: "false" 
   PHOTOPRISM_DEFAULT_TLS: "true" 

is the trailing slash appropriate? Only thing I can think of.

Maybe I'll go to a reverse proxy and get a certificate.  I'm just confused as to what I'm missing.

**Reply by lastzero** [reply]
PHOTOPRISM_INIT is only working the first time the container starts. To change the port, it's best to only change the left side eg 9002:2342 and leave the other port related  settings untouched.

**Reply by arbourp** [reply]
I don't know what was wrong along the way, but I completely restarted and things are now working as expected.

Thanks for the help!
